cmd/config/internal/commands/cmdset.go

```diff
@@ -40,6 +40,8 @@ func NewSetRunner(parent string) *SetRunner {
 	c.Flags().StringVar(&setterVersion, "version", "",
 		"use this version of the setter format")
 	c.Flags().MarkHidden("version")
+	c.Flags().BoolVar(&r.Set.ForceStringType, "force-string-type", false,
+		"force the yaml type of the value to be string")
```

Comment:
Looking at this message it looks like Alice can force a value to be string even if Charlie wants it to be an integer. Is that expected ?

Reply:
In this case we are just taking Alice's intention. Alice can't force string against Charlie's intent. So how about just flag `--type`?

So here are the cases
1. Charlie specified type to be string, Alice tried to set it to 1234 but failed. Then Alice uses flag --type=string and it should succeed.
2. Charlie specified type to be integer, Alice tried to set it to `foo` but failed. Then Alice uses flag --type=integer and it should still fail as it is easy to determine for us that foo is not integer.
3. Charlie specified type to be boolean, Alice tried to set it to `yes` but failed. Then Alice uses flag --type=boolean and it should succeed as yes is valid bool for yaml.

Reply:
If that is the case, I am not sure if we should encourage that as the setter definition says it should be integer but the actual value is string. Alice should not override Charlie's intent IMO. The only way for Alice to override Charlie's intent is to take role of Charlie and modify Kptfile to set the desired type in Kptfile.

Other tools like Terraform doesn't support overriding publisher's intent. Only way is to modify the rule.
https://www.terraform.io/docs/configuration/variables.html#custom-validation-rules

Reply:
The types are there in order to help users, not for enforcement. There are many ways a user can go around the type restrictions in the setters/schema, so I don't think there is any need to make it difficult.
If users need enforcement, they should handle it through validation with kpt functions in their CD pipeline.